File: CS50x/pset6/mario/more/mario.py
from cs50 import get_int

# recursive function to print pyramids


def printpyr(h, max):

    # base case
    if h == 0:
        return

    printpyr(h - 1, max)

    # print leading spaces
    for x in reversed(range(max - h)):
        print(" ", end='')

    # print hashes for first pyramid
    for x in range(h):
        print("#", end='')

    # print mid spaces
    print("  ", end='')

    # print second pyramid
    for x in range(h):
        print("#", end='')

    # skip to next line
    print()


# "do while loop" to check inputs
while True:
    height = get_int("Height: ")
    if (height > 0 and height < 9):
        break
    print("Invalid height.")

printpyr(height, height)

File: CS50x/pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height = 0;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    for (int i = 1; i <= height; i++)
    {
        // print spaces descending from height - 1 to 0
        for (int j = height - i; j > 0; j--)
        {
            printf(" ");
        }

        // print # for left pyramid
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }

        // print spaces in between
        printf("  ");

        // print right pyramid
        for (int m = 0; m < i; m++)
        {
            printf("#");
        }

        printf("\n");
    }

}